### ISSUE TITLE
Import dict rather than sequence? e.g. Instance(store, module, {"env": Memory(store, MemoryType(Limits(5, 5)))}) 

### ISSUE THREAD

**Issue by INPUTrrr0** [bytecodealliance/wasmtime-py#240]
Why is the import in Instance defined as "imports: Sequence[AsExtern]" rather than a dict? So instead of having _(import "" "hello")_ in hello.wat, I can have (import "env" "hello"). This would make the import more organized. 
(Although linker make use of that first slot in the import, in my understanding it's more like for linking two .wat file files, rather than organizing the imports? Or am I just not understanding how linker works. Please let me know :) ) 

Thanks!

**Comment by alexcrichton** [comment]
Yeah for this it's recommended to use `Linker` since that does name-based resolution for host functions. Would that work for your use case?

**Comment by INPUTrrr0** [comment]
Thanks Alex. I see. Previously my understanding is that for `Linker` to be used, there must be another module to import. But I only want to give it a different name for clarity of the code. But I can take a closer look at the code.

**Comment by alexcrichton** [comment]
Ah yes you can also use [`define_func`](https://bytecodealliance.github.io/wasmtime-py/#wasmtime.Linker.define_func) to add host functions into a `Linker` too

**Comment by INPUTrrr0** [comment]
Hi Alex. Thanks for your reply. I will take a look at it. I have a follow up question regarding import. I want to import a function with no input parameters. However the code below seems to indicate that the input parameters cannot be none. And The ValType doesn't seem to have a "None" type either. Thanks!
```
class FuncType(Managed["ctypes._Pointer[ffi.wasm_functype_t]"]):
    def __init__(self, params: List[ValType], results: List[ValType]):
        for param in params:
            if not isinstance(param, ValType):
                raise TypeError("expected ValType")
        for result in results:
            if not isinstance(result, ValType):
                raise TypeError("expected ValType")
```
